lib/matplotlib/collections.py

```diff
@@ -899,6 +900,118 @@ def set_paths(self, paths):
     def get_paths(self):
         return self._paths
 
+    def legend_items(self, mode="colors", useall="auto", num=10,
+                     fmt=None, func=lambda x: x, **kwargs):
+        """
+        Creates legend handles and labels for a PathCollection. This is useful
+        for obtaining a legend for a :meth:`~.Axes.scatter` plot. E.g.::
+
+            scatter = plt.scatter([1,2,3], [4,5,6], c=[7,2,3])
+            plt.legend(*scatter.legend_items())
+
+        Also see the :ref:`automatedlegendcreation` example.
+
+        Parameters
+        ----------
+        mode : string, optional, default *"colors"*
+            Can be *"colors"* or *"sizes"*. In case of *"colors"*, the legend
+            handles will show the different colors of the collection. In case
+            of "sizes", the legend will show the different sizes.
+        useall : bool or string "auto", optional (default "auto")
```

Comment:
EDIT: oops, this somehow got lost: I don't think you need both `useall` and `num` kwargs.  num=None is all data, num=int is number bins, num='auto' is auto

Reply:
I *think* that won't work, but I will recheck.

Reply:
No, you're right, that works. The only "drawback" is that in that case `num` cannot act as a parameter for the automatic determination; but I guess it's fine to have that parameter hardcoded (choosing num=9 now for the border between showing all data or some cross section of it).

Reply:
I didn't get that num interacted w/ auto somehow.  (still don't get it w/o checking the code).

I wonder if what you really want is a "bins" argument, and make this completely analagous to the histogram bins argument.  https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html  You could even optionally put in the legend how many elements there are in each bin.

Reply:
I don't think I want "bins". I want "ticks"; and I get them using a Locator. From my side this is all working fine. The idea is of course to get nice "tick positions", unlike in the case of histogram bins where you'd end up with arbitrarily ugly numbers like 1.57214.

What this function does not allow for is to add custom ticks (which is possible for bins in `histogram`). I decided against that because I think this is exactly the point where you'd anyways simply create the legend the way you want it with the artists created manually. But if someone feels that this is needed, I could also add it.

Reply:
OK, thats cool.  What about letting the user pass in a Locator?

Reply:
That is sure possible. I think it would then simply ignore `num` completely. I'm not sure if it's needed, but if there is the wish to add it, I can do that.